### PR TITLE
Added bintray and sbt-git publish settings

### PR DESCRIPTION
**Pull request by corruptmemory**
Added plugins and settings for bintray (and sbt-git).

/cc @jsuereth

### CODE REVIEW

**Comment by corruptmemory** [comment]
Also /cc @benmccann

**Comment by benmccann** [comment]
Looks fine to me. I don't know the publishing code very well though. Do we still need this line if we're publishing to bintray?

    val uri = "https://private-repo.typesafe.com/typesafe/ivy-" + id

File: version.sbt [hunk 1]
```diff
@@ -1 +1,3 @@
-version in ThisBuild := "4.0.0-SNAPSHOT"
+enablePlugins(GitVersioning)
```
Comment (jsuereth) [inline version.sbt]: sbt eclipse also, for some reason, has TWO projects.  YOu probably need to do this in both projects.

**Comment by corruptmemory** [comment]
@benmccann - no, oversight.

**Comment by corruptmemory** [comment]
@jsuereth - fix for sbt-git acceptable?

**Comment by jsuereth** [comment]
Yeah, the fix is fine, although I like the autoplugin option better :(.  Stil LGTM

File: project/Build.scala [hunk 2]
```diff
@@ -65,6 +68,9 @@ object Build extends Build {
       sbtDependency in GlobalScope := {
         (sbtDependency in GlobalScope).value.copy(revision = (sbtVersion in GlobalScope).value)
       },
+      publishMavenStyle := false,
+      bintrayOrganization in bintray := None,
```
Comment (jsuereth) [inline project/Build.scala]: This should be `bintrayOrganization in bintray := Some("sbt")`

Comment (jsuereth) [inline project/Build.scala]: I think you also want:

`name in bintray := "sbteclipse"`

**Pull request merged**